lptree.pas

```diff
@@ -5730,6 +5865,9 @@ function TLapeTree_For.Compile(var Offset: Integer): TResVar;
   Assert(FCounter <> nil);
   Assert(FLimit <> nil);
 
+  if WalkIn then
+    Exit( CompileForIn(Offset) );
```

Comment:
It looks like `for..in` is compiled significantly different from `for..to`; maybe we can move this to a separate class, e.g. `TLapeTree_ForIn`?